prepare/metrics/rag.py

```diff
@@ -174,4 +176,120 @@
 for metric_id, metric in metrics.items():
     add_to_catalog(metric, metric_id, overwrite=True)
 
-add_to_catalog(metric, "metrics.token_overlap_with_context", overwrite=True)
+# rag metrics:
+
+# reference-less:
+#   context-relevance:
+#       metrics.rag.context_relevance
+#   faithfulness:
+#       metrics.rag.faithfulness
+#       metrics.rag.k_precision
+#       metrics.rag.bert_k_precision
+#   answer-relevance:
+#       metrics.rag.answer_relevance
+
+# reference-based:
+#   context-correctness:
+#       metrics.rag.mrr
+#       metrics.rag.map
+#       metrics.rag.precision@K
+#   answer-correctness:
+#       metrics.rag.correctness
+#       metrics.rag.recall
+#       metrics.rag.bert_recall
+
+for metric_name in ["map", "mrr"]:
+    metric = MetricPipeline(
+        main_score="score",
+        preprocess_steps=[
+            CopyFields(field_to_field=[("context_ids", "prediction")], use_query=True),
+            CopyFields(
+                field_to_field=[("ground_truths_context_ids", "references")],
+                use_query=True,
+            ),
+        ],
+        metric=f"metrics.{metric_name}",
+    )
+    add_to_catalog(metric, f"metrics.rag.{metric_name}", overwrite=True)
+
+
+context_relevance = MetricPipeline(
+    main_score="score",
+    preprocess_steps=[
+        CopyFields(field_to_field=[("contexts", "references")], use_query=True),
+        CopyFields(
+            field_to_field=[("question", "prediction")],
+            use_query=True,
+        ),
+    ],
+    metric="metrics.perplexity_q.flan_t5_small",
+)
+add_to_catalog(context_relevance, "metrics.rag.context_relevance", overwrite=True)
+
+for new_catalog_name, base_catalog_name in [
+    ("metrics.rag.faithfulness", "metrics.token_overlap"),
+    ("metrics.rag.k_precision", "metrics.token_overlap"),
```

Comment:
```suggestion
    ("metrics.rag.precision_at_k", "metrics.token_overlap"),
```

Reply:
a bit clearer IMO. btw, how is the K determined?

Reply:
or maybe this isn't standard precision_at_k? i see in the comment above `metrics.rag.precision@K` is also listed

Reply:
it's "knowledge precision" - so k_precision is the common term for this.